### PR TITLE
model and testing notebook

### PR DESCRIPTION
**Pull request by colinwinslow**

### CODE REVIEW

File: python/lsst/ts/pressureSensor/Untitled.ipynb [hunk 1]
```diff
@@ -0,0 +1,120 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tm = TransducerModel(\"10.0.1.125\",502)"
```
Comment (patrickingraham) [inline python/lsst/ts/pressureSensor/Untitled.ipynb]: See Tiago's comment on ticket on where notebook should be placed if it's an example that is intended for end-users (if it's a personal test then it shouldn't be committed)... examples are great!
Comments need to be added to assist the user in what might need changing (IPs etc) and any other items of interest.

File: python/lsst/ts/pressureSensor/model.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+from pymodbus.client.sync import ModbusTcpClient as ModbusClient
+from pymodbus.exceptions import ConnectionException, ModbusIOException
+
+
+class TransducerModel():
+    """
+    Class that reads the transducer voltage through an ADAM 6024 device
```
Comment (patrickingraham) [inline python/lsst/ts/pressureSensor/model.py]: this really reads any voltage and is more broadly applicable. So maybe just say the pressure transducer is an example of an application.

File: python/lsst/ts/pressureSensor/model.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    def __init__(self, ip, port):
+        self.client = None
+        self.clientip = ip
+        self.clientport = port
+
+    def connect(self):
+        self.client = ModbusClient(self.clientip, self.clientport)
+
+    def _readVoltage(self):
+        """ reads the voltage off of ADAM-6024's inputs for channels 0-3.
```
Comment (patrickingraham) [inline python/lsst/ts/pressureSensor/model.py]: Why only 4? Should explain any limitations in the code.

File: python/lsst/ts/pressureSensor/model.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        ----------
+        counts : integer
+            16-bit integer received from ADAM device
+
+        Returns
+        -------
+        volts : float
+            counts converted into voltage number
+        """
+        ctv = 20/65535
```
Comment (patrickingraham) [inline python/lsst/ts/pressureSensor/model.py]: what is this number? where does it come from?

File: python/lsst/ts/pressureSensor/model.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        return counts * ctv - 10
```
Comment (patrickingraham) [inline python/lsst/ts/pressureSensor/model.py]: why the -10 ?
Looks like you should also have a blank line after this.

File: python/lsst/ts/pressureSensor/model.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+        Returns
+        -------
+        volts : List of floats
+            the voltages on the ADAM's first 4 input channels
+        """
+        try:
+            readout = self.client.read_input_registers(0, 8, unit=1)
+            return [self._countsToVolts(r) for r in readout.registers]
+        except AttributeError:
```
Comment (patrickingraham) [inline python/lsst/ts/pressureSensor/model.py:51]: This is great! Useful information. I'm glad you put it here.

**Review by patrickingraham — CHANGES_REQUESTED** [review]

**Comment by tribeiro** [comment]
Note that python naming convention is that method should be lower case:

https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html#methods

`_readVoltage` -> `_read_voltage`
`_countsToVolts` -> `_counts_to_volts`

Also why `_readVoltage` and `_countsToVolts` are defined as private (e.g. why the leading underscore)? Seems to me they should be regular class methods. 

I would recommend renaming

`_readVoltage` -> `read_voltage`
`_countsToVolts` -> `counts_to_volts`

File: python/lsst/ts/pressureSensor/model.py (hunk 4 above)
Comment (tribeiro) [reply python/lsst/ts/pressureSensor/model.py]: Seems like this is some sort of gain? I imagine you want this as a configurable class attribute.

File: python/lsst/ts/pressureSensor/model.py (hunk 5 above)
Comment (tribeiro) [reply python/lsst/ts/pressureSensor/model.py]: Looks like the `-10` is the "zero" of the scale. Should also be a configurable class attribute.

File: python/lsst/ts/pressureSensor/model.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        client : ModbusClient
+            the pymodbus object representing the ADAM 6024
+    """
+
+    def __init__(self, ip, port):
+        self.client = None
+        self.clientip = ip
+        self.clientport = port
+
+    def connect(self):
```
Comment (tribeiro) [inline python/lsst/ts/pressureSensor/model.py:30]: should there be a `disconnect` method as well?

File: python/lsst/ts/pressureSensor/model.py (hunk 4 above)
Comment (colinwinslow) [reply python/lsst/ts/pressureSensor/model.py]: This is just the way the ADAM devices discretize voltages. It doesn't directly provide a float but instead provides an int that can indicate a voltage anywhere between -10 and +10. It won't change unless we use different hardware. And if we use different hardware, probably a lot of things would need to change.

File: python/lsst/ts/pressureSensor/model.py (hunk 5 above)
Comment (colinwinslow) [reply python/lsst/ts/pressureSensor/model.py]: Made the 20 and 10 into class attributes. The 65535 should never change as long as we're using ADAM devices.

Comment (patrickingraham) [reply python/lsst/ts/pressureSensor/model.py]: you now have      self.range_start = -10 but the last line is "return counts * ctv - self.range_start"

I think you have a negative sign problem here. The new code adds 10, the first version subracts 10...

**Comment by patrickingraham** [comment]
Much better. Found a negative sign issue I think, please clarify/fix before merging.

**Review by patrickingraham — APPROVED** [review]
Please fix negative sign issue before merging (see comments)

**Pull request merged**